kernel/src/log_replay.rs

```diff
@@ -0,0 +1,179 @@
+use crate::{
+    actions::deletion_vector::DeletionVectorDescriptor,
+    engine_data::{GetData, TypedGetData},
+    DeltaResult,
+};
+use std::collections::HashSet;
+use tracing::debug;
```

Comment:
nit: grouped sorting (and not as much nesting)
```suggestion
use std::collections::HashSet;

use crate::actions::deletion_vector::DeletionVectorDescriptor;
use crate::engine_data::{GetData, TypedGetData};
use crate::DeltaResult;

use tracing::debug;
```

Reply:
(I realize these may have just been moved over from elsewhere but I'm generally trying to get us in a better habit of import hygeine)

Reply:
Thanks for pointing out! All good, I should build the habit 👍